steps/rabbit/session.go

```diff
@@ -0,0 +1,334 @@
+// Copyright 2021 Telefonica Cybersecurity & Cloud Tech SL
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rabbit
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/Telefonica/golium"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/streadway/amqp"
+	"github.com/tidwall/sjson"
+)
+
+// Session contains the information of a rabbit session.
+type Session struct {
+	Connection *amqp.Connection
+	// Messages received from the publish/subscribe channel
+	Messages []amqp.Delivery
+	// Correlator is used to correlate the messages for a specific session
+	Correlator string
+	// rabbit channel
+	// It is stored after subscription to close the subscription
+	channel *amqp.Channel
+	// rabbit subscription channel
+	subCh <-chan amqp.Delivery
+	// rabbit headers to store specific data
+	headers amqp.Table
+	// rabbit publishing message
+	publishing amqp.Publishing
+	// rabbitmq received delivery message
+	msg amqp.Delivery
+}
```

Comment:
Just as a comment. I see that you either write `rabbit` or `rabbitmq` or `RabbitMQ`, etc., along all the code (comments and errors).
It should be good to review and unify the criteria.